ui/packages/atlasmap-core/src/utils/mapping-serializer.ts

```diff
@@ -283,7 +283,11 @@ export class MappingSerializer {
 
     let inputField = [];
     if (mappingJson.inputFieldGroup) {
-      mapping.transition.mode = TransitionMode.MANY_TO_ONE;
+      if (mappingJson.inputFieldGroup.fieldType === 'COMPLEX') {
```

Comment:
Can we just use TransitionMode.EXPRESSION whenever if expression is enabled? Is there a reason to distinguish the "expression consumes COMPLEX field" from other expression as a separate state?

Reply:
I got a question related to this - what do we show if expression is enabled now? if it's MANY_TO_ONE, does it show a dropdown for many-to-one transformation drondown in Mapping Details? If so I think that's a bug, many-to-one transformation should not be selectable for expression mapping.

Reply:
Since a complex field reference doesn't appear in the mapping model as a field (it's an attr of the input field group) we need to distinguish a complex reference from a simple mapping where someone enable the expression box.

Reply:
> I got a question related to this - what do we show if expression is enabled now? if it's MANY_TO_ONE, does it show a dropdown for many-to-one transformation drondown in Mapping Details? If so I think that's a bug, many-to-one transformation should not be selectable for expression mapping.

The transformation dropdown is disabled in the expression box is enabled.

Reply:
So TransitionMode doesn't make any sense if expression is enabled isn't it? anyway, the state "if there's at least one COMPLEX field in expression" doesn't fit in TransitionMode, maybe mapping.transition.expression.hasComplexField or something?

Reply:
Removed COMPLEX_EPXR from transition mode in favor of hasCOmplexField expression boolean.